controller/src/main/java/io/pravega/controller/store/stream/Stream.java

```diff
@@ -196,39 +200,102 @@
     CompletableFuture<List<Long>> getActiveSegments(int epoch);
 
     /**
-     * Called to start metadata updates to stream store wrt new scale event.
      *
+     * @return
+     */
+    CompletableFuture<VersionedMetadata<EpochTransitionRecord>> getEpochTransition();
+
+    /**
+     * Called to start metadata updates to stream store with respect to new scale request. This method should only update
```

Comment:
This is good info, I still feel it might nice to update to update the javadoc comment with a longer description of how it is used with the different flows (manual, auto), using the info you provided above. The one problem I see with extending the description to talk specifics of the current processing is making the description very targeted to the current way of processing requests. In some sense, it might reduce the generality of the method, which might not matter as it is an internal method. My ultimate goal is any case more clarity across the code base.

Reply:
> The one problem I see with extending the description to talk specifics of the current processing is making the description very targeted to the current way of processing requests. 

Yes, we would want to keep the comment general enough and focus on expectation for the caller while not describing about the internal metadata structures. 

I think the processing specific information is encapsulated in the comments in the code ScaleOperationTask and that is the correct place for it.